packages/@react-spectrum/menu/src/SafeTriangle.tsx

```diff
@@ -20,66 +20,54 @@ interface SafeTriangleProps {
 }
 
 /**
- * Allows the user to move their pointer to the sub-menu without it closing
- * due to the mouse leaving the trigger element.
+ * Allows the user to move their pointer to the sub-menu without it closing when their mouse leaves the trigger element.
+ * Renders an invisible triangle from the mouse position to the inside corners of the sub-menu.
  */
 export function SafeTriangle(props: SafeTriangleProps): ReactElement {
   let {
     subMenuRef,
     triggerRef
   } = props;
-  let [mousePosition, setMousePosition] = useState<{mouseX: number, mouseY: number}>({mouseX: null, mouseY: null});
-  let {mouseX, mouseY} = mousePosition;
-  let updateMousePosition = (e: MouseEvent) => {
-    setMousePosition({mouseX: e.clientX, mouseY: e.clientY});
-  };
+  let [style, setStyle] = useState<React.CSSProperties>({height: 0, width: 0});
 
   useEffect(() => {
-    window.addEventListener('mousemove', updateMousePosition);
+    let onMouseMove = (e: MouseEvent) => {
+      let mouseX = e.clientX;
+      let mouseY = e.clientY;
+      if (subMenuRef.current && triggerRef.current) {
+        let {left, right, top, height} = (subMenuRef.current?.UNSAFE_getDOMNode() as HTMLElement).getBoundingClientRect();
+        let offset = triggerRef.current?.getBoundingClientRect().width;
+        let direction = left > triggerRef.current?.getBoundingClientRect().left ? 'right' : 'left';
 
-    return () => window.removeEventListener('mousemove', updateMousePosition);
-  }, []);
+        let isHidden = (direction === 'right' && mouseX > left) || (direction === 'left' && mouseX < right);
 
-  let x = 0;
-  let y = 0;
-  let width = 0;
-  let height = 0;
-  let direction: 'left' | 'right';
-  let subMenuNode = subMenuRef.current?.UNSAFE_getDOMNode() as HTMLElement;
-  if (subMenuNode) {
-    let rect = subMenuNode.getBoundingClientRect();
-    x = rect.left;
-    y = rect.top;
-    width = mouseX > x ? 0 : x - mouseX + 5;
-    height = rect.height;
-    direction = x > triggerRef.current?.getBoundingClientRect().left ? 'right' : 'left';
-  } else {
-    return null;
-  }
+        setStyle(isHidden ? {
+          height: 0,
+          width: 0
+        } : {
+          top: triggerRef.current?.getBoundingClientRect().top - triggerRef.current?.parentElement.getBoundingClientRect().top,
+          left: direction === 'right' ? mouseX - left + offset : undefined,
+          height,
+          width: direction === 'right' ? left - mouseX : mouseX - right,
+          clipPath: direction === 'right' ? `polygon(100% 0%, 0% ${(100 * (mouseY - top)) / height}%, 100% 100%)` : `polygon(0% 0%, 100% ${(100 * (mouseY - top)) / height}%, 0% 100%)`
+        });
+      }
+    };
+  
+    window.addEventListener('mousemove', onMouseMove);
```

Comment:
pointermove is preferred, if it should only be for mouse, you can check the event for the pointertype
not sure we still need to fall back to mousemove, pointer events are supported on all the browsers we target now